src/bentoml/_internal/marshal/dispatcher.py

```diff
@@ -363,12 +366,23 @@ async def controller(self):
                         continue
                     await asyncio.sleep(self.tick_interval)
                     continue
-                if (
+
+                # we are now free to dispatch whenever we like
+                while (
+                    # if we don't already have enough requests,
                     n < self.max_batch_size
-                    and n * (wn + dt + (a or 0)) <= self.optimizer.wait * decay
+                    # we are not about to cancel the first request,
+                    and latency_0 + dt <= self.max_latency_in_ms * 0.95
```

Comment:
They're different things; this is just to prevent requests from going past max latency so I put in an arbitrary limiter, the other is obviously wait time delay. I actually think delay should probably be something like .5, but that change is for another day.